app/views/accounts/_account_list.html.erb

```diff
@@ -3,29 +3,29 @@
 <% accounts = Current.family.accounts.where(accountable_type: type.name) %>
 
 <% if accounts.sum(&:converted_balance) > 0 %>
-<details class="mb-1 text-sm group">
-  <summary class="flex gap-4 px-2 py-3 items-center w-full rounded-[10px] font-medium hover:bg-[#f2f2f2]">
-    <%= lucide_icon("chevron-down", class: "hidden group-open:block text-[#737373] w-5 h-5") %>
-    <%= lucide_icon("chevron-right", class: "group-open:hidden text-[#737373] w-5 h-5") %>
-    <div class="text-left"><%= type.model_name.human %></div>
-    <div class="ml-auto"><%= format_currency accounts.sum(&:converted_balance) %></div>
-  </summary>
-  
-  <% accounts.each do |account| %>
-    <%= link_to account_path(account), class: "flex items-center w-full gap-3 px-2 py-3 mb-1 hover:bg-[#f2f2f2] rounded-[10px]" do %>
-      <div>
-        <p class="font-medium"><%= account.name %></p>
-        <% if account.subtype %>
-          <p class="text-xs text-[#737373]"><%= account.subtype&.humanize %></p>
-        <% end %>
-      </div>
-      <p class="ml-auto font-medium"><%= format_currency account.converted_balance %></p>
+  <details class="mb-1 text-sm group" data-controller="account-list" data-account-list-category-value="<%= type.name %>" data-action="toggle->account-list#handleToggle"> 
```

Comment:
It might be better to make a more generic stimulus controller that persists the state of any details element rather than specifically just the account list.

Perhaps we could give the details element an id, and use that as it's key?

The name of the action could also be more descriptive, perhaps
`toggle->details-state-controller#persistState`?

Reply:
Good point, udpated!